lib/rust/ensogl/core/src/display/navigation/navigator/events.rs

```diff
@@ -326,8 +333,10 @@ impl NavigatorEvents {
                         MovementType::Zoom { focus } => {
                             let zoom_speed = data.zoom_speed();
                             let zoom_amount = movement_to_zoom(movement);
-                            let zoom_event = ZoomEvent::new(focus, zoom_amount, zoom_speed);
-                            data.on_zoom(zoom_event);
+                            if let Some(zoom_event) = ZoomEvent::new(focus, zoom_amount, zoom_speed)
+                            {
```

Comment:
```suggestion
                            if let Some(zoom_event) = ZoomEvent::new(focus, zoom_amount, zoom_speed) {
```

Reply:
This is `rustfmt`, not me! Looks weird indeed, I changed variable name from `zoom_event` to `event` to fix it.